internal/errcheck/errcheck_test.go

```diff
@@ -5,10 +5,14 @@ import (
 	"go/build"
 	"go/parser"
 	"go/token"
+	"regexp"
 	"testing"
 )
 
-const testPackage = "github.com/kisielk/errcheck/testdata"
+const (
+	testPackage       = "github.com/kisielk/errcheck/testdata"
+	testVendorPackage = "github.com/kisielk/errcheck/testvendor"
```

Comment:
I don't really like that `testvendor` is separate from `testdata`, but it's not possible to run the vendor test inside `testdata` because Go imports doesn't do vendoring inside of a `testdata` directory (this is a restriction that's hard-coded in core Go code).

Options are:
1. Keep as is (or rename the `testvendor` directory to something else)
2. Remove the unit test for checking the vendor directory
3. Change test so that vendor package is dynamically written to temporary directory as part of test (either copied from `testdata` or written from string constants).

Let me know what you think is most preferable and happy to update PR to do so.